blog/nse_portfolio_sorts/index.qmd

```diff
@@ -0,0 +1,931 @@
+      mutate(sorting_month = month)
+  }
+  
+  # Double independent sort
+  if(sorting_method == "dbl_ind") {
+    # Assign portfolios
+    data_sorting <- data_sorting |> 
+      sort_double_ind(sorting_variable = sorting_variable, 
+                      exchanges = exchanges, 
+                      n_portfolios_main = n_portfolios_main,
+                      n_portfolios_secondary = n_portfolios_secondary) |> 
+      select(permno, month, portfolio, portfolio_main, portfolio_secondary) |> 
+      mutate(sorting_month = month)
+  }
+    
+  # Double dependent sort
+  if(sorting_method == "dbl_dep") {
+    # Assign portfolios
+    data_sorting <- data_sorting |> 
+      sort_double_dep(sorting_variable = sorting_variable, 
+                      exchanges = exchanges, 
+                      n_portfolios_main = n_portfolios_main,
+                      n_portfolios_secondary = n_portfolios_secondary)  |> 
+      select(permno, month, portfolio, portfolio_main, portfolio_secondary) |> 
+      mutate(sorting_month = month)
+  }
+  
+  # Compute portfolio return
+  if(sorting_method == "single") {
+    data |>
+      left_join(data_sorting, by = c("permno", "month")) |>
+      group_by(permno) |>
+      arrange(month) |>
+      fill(portfolio, sorting_month) |>
+      filter(sorting_month >= month %m-% months(12)) |>
+      drop_na(portfolio) |>
+      group_by(month, portfolio) |>
+      summarize(ret = if_else(value_weighted, weighted.mean(ret_excess, mktcap_lag), mean(ret_excess)),
+                .groups = "drop")
+  } else {
+    data |>
+      left_join(data_sorting, by = c("permno", "month")) |>
+      group_by(permno) |>
+      arrange(month) |>
+      fill(portfolio_main, portfolio_secondary, portfolio, sorting_month) |>
+      filter(sorting_month >= month %m-% months(12)) |>
+      drop_na(portfolio_main, portfolio_secondary) |>
+      group_by(month, portfolio) |>
+      summarize(ret = if_else(value_weighted, weighted.mean(ret_excess, mktcap_lag), mean(ret_excess)),
+                portfolio_main = unique(portfolio_main),
+                .groups = "drop") |>
+      group_by(month, portfolio_main) |>
+      summarize(ret = mean(ret),
+                .groups = "drop") |> 
+      rename(portfolio = portfolio_main)
+  }
+}
+```
+
+### Combining the functions
+
+Now, we have everything to compute our 69,120 portfolio sorts for *asset growth* to understand the variation our decisions induce. To achieve this, our function considers all choices as arguments and passes them to the sample selection and portfolio construction functions.
+
+Finally, the function computes the return differential for each month. Since we are only interested in the mean here, we simply take the mean of these time series and call it our premium estimate. Notice that we also multiply the estimate by `-100` because the sorting variable *asset growth* predicts slow-growing firms to outperform, i.e., our portfolio sort would go in the wrong direction.
```

Comment:
Die mulitplication by -100 macht die function nicht generic, wie du eingangs behauptest. Also man kann nicht problemlos andere sorting Variablen verwenden. Entweder du wirst weniger generic in der Beschreibung oder du gibst noch irgendeinen Parameter dazu, der die direction angibt. Allerdings fällt mir ad hoc kein guter Name für die Variable ein.

Reply:
This is a very good point. I moved it to a separate chunk and added an explanation (Commit https://github.com/tidy-finance/website/pull/34/commits/75f872b23e4d472eb2e6754d688c3f206bb6030d).